### PR TITLE
Various DBI related fixes that make it possible to load DBI on Linux

### PR DESCRIPTION
**Pull request by Djuffin**
This change make it possible to write a managed debugger on Linux that can
1. Load and initialized DBI
2. Create and setup CorDebug interface
3. Successfully call CreateProcess or DebugActiveProcess process (hr = 0)
4. then crash :)

### CODE REVIEW

**Comment by Djuffin** [comment]
cc @mikem8361

**Comment by mikem8361** [comment]
LGTM.

**Comment by mikem8361** [comment]
@jkotas - do you want to take at this?

File: src/debug/shared/dbgtransportsession.cpp [hunk 1]
```diff
@@ -1107,6 +1107,9 @@ HRESULT DbgTransportSession::CheckBufferAccess(__in_ecount(cbBuffer) PBYTE pbBuf
         return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
     }
 
+    // TODO: VirtualQuery from PAL doesn't seem to provide correct result for DAC globals on Linux.
```
Comment (jkotas) [inline src/debug/shared/dbgtransportsession.cpp]: This is by design. VirtualQuery only works well for memory allocated via VirtualAlloc.

**Comment by Djuffin** [comment]
сс @noahfalk

File: src/debug/shared/dbgtransportsession.cpp (hunk 1 above)
Comment (jkotas) [inline src/debug/shared/dbgtransportsession.cpp]: Could you please fix the comment?

Comment (Djuffin) [inline src/debug/shared/dbgtransportsession.cpp]: Sure. Done.

**Comment by jkotas** [comment]
LGTM

**Pull request merged**